src/sentry/search/events/fields.py

```diff
@@ -2265,6 +2313,26 @@ def __init__(self, dataset: Dataset, params: ParamsType):
                     ),
                     default_result_type="percentage",
                 ),
+                SnQLFunction(
+                    "to_other",
+                    required_args=[
+                        SnQLColumn("column", allowed_columns=["release", "trace.parent_span"]),
+                        SnQLStringArg("value", unquote=True, unescape_quotes=True),
+                    ],
+                    optional_args=[
+                        with_default("that", SnQLStringArg("that")),
+                        with_default("this", SnQLStringArg("this")),
+                    ],
+                    snql_column=lambda args, alias: Function(
+                        "if",
+                        [
+                            Function("equals", [args["column"], args["value"]]),
```

Comment:
Thoughts on doing something like this, so we don't need a SnQLColumn? eg.
```suggestion
                            Function("equals", [self.column(args["column"]), args["value"]]),
```

My thought is we'll only be using Columns here, so calling the full resolve field is unecessary and that way we don't need to special case the `SnQLColumn` argument

Reply:
So this is what I initially thought too, but I think it's cleaner to resolve the column [here](https://github.com/getsentry/sentry/pull/27782/files#diff-e7d5c143e6b668a98f4dc22ec4fdc3796c413d23cfbd4d3d43c1cd001465cddeR2480) so that we aren't calling self.resolve_field or self.column on column args all over the different Functions. I like the idea of resolving column args in one place for all functions.
It's also nice to maintain the pattern of validating the args in the normalize method of the FunctionArgs subclasses. In this PR we introduce `SnQLColumn` but I know we'll need one to validate AggregateAliases, Numeric Columns, Columns + Field Aliases etc. So it'd be great if we can just validate all of this in the `normalize(...)` and just call resolve_field on those args and we know it will resolve to the right thing.

Reply:
:+1: that's a good motivation